src/javascript/app_2/App/Components/Elements/SettingsDialog/settings_dialog.jsx

```diff
@@ -0,0 +1,61 @@
+import classNames      from 'classnames';
+import React           from 'react';
+import PropTypes       from 'prop-types';
+import Tabs            from '../Tabs/tabs.jsx';
+import ChartSettings   from '../../../../Modules/Trading/Containers/SettingsDialog/settings_chart.jsx';
+import GeneralSettings from '../../../../Modules/Trading/Containers/SettingsDialog/settings_general.jsx';
+import LanguageDialog  from '../../../../Modules/Trading/Containers/SettingsDialog/settings_language.jsx';
+import { localize }    from '../../../../../_common/localize';
+
+class SettingsDialog extends React.Component {
+
+    setWrapperRef = (node) => {
+        this.wrapper_ref = node;
+    }
+
+    // TO-DO - Simplify this
+    handleClickOutside = (event) => {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.props.is_open && !(event.target.className === 'ic-settings')) {
+            this.props.toggleDialog();
+        }
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    render() {
+        const settings_dialog_class = classNames('settings-dialog', {
+            'show': this.props.is_open,
+        });
+        const settings_dialog_container_class = classNames('settings-dialog-container', {
+            hide: this.props.is_language_dialog_visible,
+        });
+        return (
+            <div ref={this.setWrapperRef} className={settings_dialog_class}>
+                <div className={settings_dialog_container_class}>
+                    <span className='settings-header'>{localize('Settings')}</span>
+                    <Tabs alignment='center' list={SettingsContent} />
+                </div>
+                <LanguageDialog />
+            </div>
+        );
+    }
+}
+
+const SettingsContent = {
```

Comment:
@zakibinary 
it's better to be defined as a static get property of the component class since you're using localize function.

```
static get settings_content() {
   return {
       1: { header: localize('General'), content: GeneralSettings },
       2: { header: localize('Chart'), content: ChartSettings },
    };
}
```

Reply:
Can we use `static` in strict mode?

Reply:
@zakibinary 
yes, just it's accessible by the name of the class `SettingsDialog.setting_content`